src/test/java/server/EchoServerTest.java

```diff
@@ -1,12 +1,65 @@
 package server;
 
+
+import mocks.MockWrapper;
 import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
 import static org.junit.jupiter.api.Assertions.*;
 
 
-class EchoServerTest {
-    @Test void walkingSkeletonTest() {
-        EchoServer classUnderTest = new EchoServer();
-        assertEquals(classUnderTest.walkingSkeletonTest(), "testing 123");
+class MockWrappersTests {
+    @Test
+    public void testRunGetsAcceptsClient() throws IOException {
+        ServerSocket serverSocket = new ServerSocket(4000);
+        MockWrapper mockWrapper = new MockWrapper(serverSocket);
+        EchoServer testServer = new EchoServer();
+
+        testServer.run(mockWrapper);
+
+        assertEquals(1, mockWrapper.getNumberOfCallsToAcceptClient());
+    }
+    @Test
+    public void testRunCreatesInputStream() throws IOException {
+        ServerSocket serverSocket = new ServerSocket(4444);
+        MockWrapper mockWrapper = new MockWrapper(serverSocket);
+        EchoServer testServer = new EchoServer();
+
+        testServer.run(mockWrapper);
+
+        assertEquals(1, mockWrapper.getNumberOfCallsToCreateInputStream());
+    }
+    @Test
+    public void testRunCreatesOutputStream() throws IOException {
+        ServerSocket serverSocket = new ServerSocket(4001);
+        MockWrapper mockWrapper = new MockWrapper(serverSocket);
+        EchoServer testServer = new EchoServer();
+
+        testServer.run(mockWrapper);
+
+        assertEquals(1, mockWrapper.getNumberOfCallsToCreateOutStream());
+    }
+    @Test
```

Comment:
Again, watch spacing between methods/tests. (I know you're doing this on your own without a linter...gain through pain...sorry).